### PR TITLE
Consistently use `db_to_json` to convert from database values to JSON objects.

### PR DESCRIPTION
**Pull request by clokep**
~~This removes the last usages of importing json from canonicaljson in order to get the simplejson package.~~

This consistently uses the `db_to_json` helper in our storage code. So a follow-on to #7836 / #7802 and related to #7674.

### CODE REVIEW

**Comment by clokep** [comment]
There are other instances of `json.loads` being used directly in `synapse.storage` -- should we be converting all of those to `db_to_json`? It is unclear to me in what situations postgresql returns a memoryview / in what situations we get bytes back instead of strings.

**Comment by clokep** [comment]
> There are other instances of `json.loads` being used directly in `synapse.storage` -- should we be converting all of those to `db_to_json`? It is unclear to me in what situations postgresql returns a memoryview / in what situations we get bytes back instead of strings.

I decided to make this change, see 281d4de29eb776c28ce23d1c72bdeb4f1dc10e0b

**Comment by clokep** [comment]
I suspect it would be nice to have a `json_to_db` helper as well that removes whitespace and such as #7372 suggests.

File: synapse/storage/background_updates.py [hunk 1]
```diff
@@ -249,7 +249,10 @@ async def _do_background_update(self, desired_duration_ms: float) -> int:
             retcol="progress_json",
         )
 
-        progress = json.loads(progress_json)
+        # Avoid a circular import.
+        from synapse.storage._base import db_to_json
```
Comment (richvdh) [inline synapse/storage/background_updates.py:253]: hrm. It's unsatisfactory if `synapse.storage._base` has a dependency on this file somehow. This is fine as a temporary solution though.

Comment (clokep) [reply synapse/storage/background_updates.py:253]: `synapse.storage._base` imports from `synapse.storage.database`, which imports from `synapse.storage.background_updater`.

**Pull request merged**